lively.ide/studio/controls/fill.cp.js

```diff
@@ -1,41 +1,130 @@
           ];
         }
       }
     };
   }
 
-  onMouseDown () { this.deactivate(); }
+  remoteAssetChecked (checked) {
+    this.targetMorph.withMetaDo({ reconcileChanges: true }, () => {
+      this.targetMorph.imageUrl = config.defaultImage;
+    });
+    this.update();
+    const { sourceDescription, button } = this.ui;
+    if (checked) {
+      sourceDescription.readOnly = false;
+      sourceDescription.nativeCursor = 'text';
+      sourceDescription.textString = 'Enter URL';
+      sourceDescription.tooltip = '';
+      button.textAndAttributes = ['', {
+        fontFamily: '"Font Awesome 6 Free", "Font Awesome 6 Brands"',
+        fontSize: 18,
+        fontWeight: '900'
+      }];
+    } else {
+      sourceDescription.readOnly = true;
+      sourceDescription.nativeCursor = 'not-allowed';
+      sourceDescription.tooltip = 'Select an image with the button to the right or enable remote assets to enter an URL.';
+      sourceDescription.textString = 'Choose Asset';
+      button.textAndAttributes = ['', {
+        fontFamily: 'Tabler Icons',
+        fontSize: 18,
+        fontWeight: '900'
+      }];
+    }
+  }
+
+  onMouseDown () {
+    this.deactivate();
+  }
+
+  async imageButtonPressed (evt) {
+    if (evt && evt.key && evt.key !== 'Enter') return;
+    if (this.ui.remoteAssetCheck.checked) {
+      try {
+        new URL(this.ui.sourceDescription.textString);
+      } catch (e) {
+        $world.setStatusMessage('Invalid URL', StatusMessageError);
+        return;
+      }
+      this.targetMorph.withMetaDo({ reconcileChanges: true }, () => {
+        this.targetMorph.imageUrl = this.ui.sourceDescription.textString;
+      });
+    } else {
+      const assetBrowser = part(AssetBrowserPopup);
+      $world._assetBrowserPopup = assetBrowser;
+      once(assetBrowser, 'close', this, 'closeAssetBrowserPopup');
+      if ($world._assetBrowser) $world._assetBrowser.block();
+      const selectedImageUrl = await assetBrowser.activate();
+      if (selectedImageUrl) {
+        this.targetMorph.withMetaDo({ reconcileChanges: true }, () => {
+          this.targetMorph.imageUrl = selectedImageUrl;
+        });
+      }
+    }
+    this.update();
+  }
+
+  closeAssetBrowserPopup () {
+    if (!$world._assetBrowserPopup) return;
+    $world._assetBrowserPopup.close();
+    $world._assetBrowserPopup = null;
+  }
 
   focusOn (target) {
     this.targetMorph = target;
     this.ui.imageControl.visible = !!target.isImage;
+    if (target.isImage) {
+      const imageUrl = target.imageUrl;
+      this.updateImageInfo(imageUrl, true);
+    }
+
     this.models.fillColorInput.targetMorph = target;
+    this.update();
+  }
+
+  updateImageInfo (imageUrl, withCheckState = false) {
+    if (imageUrl.includes('local_projects')) {
+      this.withoutBindingsDo(() => {
+        this.ui.sourceDescription.textString = imageUrl.split('/').pop();
+        if (withCheckState) this.ui.remoteAssetCheck.checked = false;
```

Comment:
This does not seem to reliably disable the editable state of the image url input field. When I previously selected a remote URL image and then switch over to a local asset image morph, the input is still editable.